From 95a2c24fd8ae18871c3fd5bc5dc468ddf57e8ddc Mon Sep 17 00:00:00 2001
From: Rajan Vaja <rajan.vaja@xilinx.com>
Date: Thu, 18 Jun 2020 04:04:47 -0700
Subject: [PATCH 1396/1851] clk: zynqmp: Map ZYNQMP_CLK_SET_RATE_GATE with CCF

commit da85012cd4d5b4fb9760bedef99f5921eb42cb54 from
https://github.com/Xilinx/linux-xlnx.git

ZYNQMP_CLK_SET_RATE_GATE is not mapped with CCF flag. So map
ZYNQMP_CLK_SET_RATE_GATE flag with CLK_SET_RATE_GATE CCF flag.

Signed-off-by: Rajan Vaja <rajan.vaja@xilinx.com>
Reviewed-by: Jolly Shah <jolly.shah@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/clk/zynqmp/clkc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/clk/zynqmp/clkc.c b/drivers/clk/zynqmp/clkc.c
index aa6d1a29a42e..a1369a753f4e 100644
--- a/drivers/clk/zynqmp/clkc.c
+++ b/drivers/clk/zynqmp/clkc.c
@@ -395,6 +395,8 @@ static int __zynqmp_clock_get_topology(struct clock_topology *topology,
 		topology[*nnodes].type = type;
 		flag = FIELD_GET(CLK_TOPOLOGY_FLAGS, response->topology[i]);
 		topology[*nnodes].flag = 0;
+		topology[*nnodes].flag |= (flag & ZYNQMP_CLK_SET_RATE_GATE) ?
+					  CLK_SET_RATE_GATE : 0;
 		topology[*nnodes].flag |= (flag & ZYNQMP_CLK_SET_RATE_PARENT) ?
 					   CLK_SET_RATE_PARENT : 0;
 		topology[*nnodes].flag |= (flag & ZYNQMP_CLK_IGNORE_UNUSED) ?
-- 
2.31.1

